### PR TITLE
chore(warehouse): warehouse integration tests

### PR DESCRIPTION
**Pull request by achettyiitr**
# Description

**Warehouse Integration Test Changes**

1. Single docker-compose file to spin up corresponding containers.
2. Separating out the core and warehouse-specific integration tests.
3. Added Integration test for RS, SNOWFLAKE, and DELTALAKE.
4. Added GA for Warehouse Service Integration test.
5. Added Parallelisation support for the CLICKHOUSE and CLICKHOUSE Cluster Integration test.
6. Remove the health check code basically (wait For eternity) which we did for Transformer and MINIO containers.
7. Code cleanup around existing Integration test changes.
8. Using a simplified version of template.json

**Corresponding Spec:**  https://www.notion.so/rudderstacks/Integration-tests-revamp-In-progress-69282d4a095047268e6163ce1cb1dcc3

## Notion Ticket

https://www.notion.so/rudderstacks/Warehouse-Integration-Tests-aceb9911fe574e62a3f3e6037947fd3f

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -63,3 +65,24 @@ lint: fmt
...
+	docker-compose -f warehouse/docker-compose.test.yml down --remove-orphans --volumes
+
+setup-warehouse-integration: cleanup-warehouse-integration
+	docker-compose -f warehouse/docker-compose.test.yml up --build start_warehouse_integration
+
+logs-warehouse-integration:
+	docker logs wh-backend
+
+run-warehouse-integration: setup-warehouse-integration
+	if docker-compose -f warehouse/docker-compose.test.yml exec -T wh-backend go test -v ./warehouse/... -tags=warehouse_integration -timeout 30m -count 1; then \
```
Comment (achettyiitr) [inline Makefile:79]: Verify if -p factor is working.

Comment (achettyiitr) [reply Makefile:79]: After removing it kind increased the test time. Currently keeping this value as 8.

**Review by Jayachand — APPROVED** [review]
lGTM

**Review by abhimanyubabbar — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**